chrome.py

```diff
@@ -21,28 +27,50 @@ def __init__(self, error):
 def alfred_error(error):
     alfred.write(alfred.xml([ErrorItem(error)]))
 
-def copy_history(profile):
-    if os.path.isfile(HISTORY_CACHE) and time.time() - os.path.getmtime(HISTORY_CACHE) < HISTORY_CACHE_EXPIRY:
-        return HISTORY_CACHE
+def copy_db(name, profile):
+    cache = os.path.join(alfred.work(True), name)
+    if os.path.isfile(cache) and time.time() - os.path.getmtime(cache) < CACHE_EXPIRY:
+        return cache
 
-    history_file = os.path.join(os.path.expanduser(profile), HISTORY_DB)
+    file = os.path.join(os.path.expanduser(profile), name)
     try:
-        shutil.copy(history_file, HISTORY_CACHE)
+        shutil.copy(file, cache)
     except:
         raise IOError(u'Unable to copy Google Chrome history database from {}'.format(history_file))
 
-    return HISTORY_CACHE
+    return cache
+
 
 def history_db(profile):
-    history = copy_history(profile)
+    history = copy_db(HISTORY_DB, profile)
+    favicons = copy_db(FAVICONS_DB, profile)
     db = sqlite3.connect(history)
+    db.cursor().execute('ATTACH DATABASE ? AS favicons', (favicons,)).close()
     return db
 
+
+def cache_favicon(image_data, uid, last_updated):
+    cache_dir = os.path.join(alfred.work(True), FAVICONS_CACHE)
+    if not os.path.exists(cache_dir):
+        os.makedirs(cache_dir)
```

Comment:
Why do we need `os.makedirs` over `os.mkdir` here? Just in case we ever want to nest the cache? I'm not sure we need to worry about that, but I'm also not sure the choice makes a huge difference either way here...